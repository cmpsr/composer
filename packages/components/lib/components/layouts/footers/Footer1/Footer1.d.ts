import { ReactElement } from "react";
import { FooterStyle } from "types";
export declare const FOOTER1_DEFAULT_TEST_ID = "footer1";
declare type Props = {
    copy?: string;
    className?: string;
    children?: {
        logo?: ReactElement;
        content?: ReactElement | ReactElement[];
    };
    style: FooterStyle;
    testId?: string;
};
export declare const Footer1: ({ className, copy, children, style, testId, }: Props) => JSX.Element;
export {};
//# sourceMappingURL=Footer1.d.ts.map