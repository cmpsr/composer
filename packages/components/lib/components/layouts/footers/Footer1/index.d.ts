export * from './Footer1';
//# sourceMappingURL=index.d.ts.map