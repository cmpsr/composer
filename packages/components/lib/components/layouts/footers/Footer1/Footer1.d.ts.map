{"version":3,"file":"Footer1.d.ts","sourceRoot":"","sources":["../../../../../src/components/layouts/footers/Footer1/Footer1.tsx"],"names":[],"mappings":"AAAA,OAAc,EAAE,YAAY,EAAE,MAAM,OAAO,CAAC;AAM5C,OAAO,EAAE,WAAW,EAAE,MAAM,OAAO,CAAC;AAEpC,eAAO,MAAM,uBAAuB,YAAY,CAAC;AAEjD,aAAK,KAAK,GAAG;IACX,IAAI,CAAC,EAAE,MAAM,CAAC;IACd,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,QAAQ,CAAC,EAAE;QACT,IAAI,CAAC,EAAE,YAAY,CAAC;QACpB,OAAO,CAAC,EAAE,YAAY,GAAG,YAAY,EAAE,CAAC;KACzC,CAAC;IACF,KAAK,EAAE,WAAW,CAAC;IACnB,MAAM,CAAC,EAAE,MAAM,CAAC;CACjB,CAAC;AAEF,eAAO,MAAM,OAAO,kDAMjB,KAAK,gBA4BP,CAAC"}