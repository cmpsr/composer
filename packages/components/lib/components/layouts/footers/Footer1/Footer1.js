"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Footer1 = exports.FOOTER1_DEFAULT_TEST_ID = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _containers = require("../../../containers");

var _utils = require("../../../../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const FOOTER1_DEFAULT_TEST_ID = "footer1";
exports.FOOTER1_DEFAULT_TEST_ID = FOOTER1_DEFAULT_TEST_ID;

const Footer1 = ({
  className,
  copy,
  children,
  style,
  testId = FOOTER1_DEFAULT_TEST_ID
}) => {
  const colorClasses = (0, _utils.getClassesFromStyle)(style.background);
  return /*#__PURE__*/_react.default.createElement(_containers.Box, {
    size: "full",
    className: (0, _classnames.default)("py-8", className, colorClasses),
    testId: testId
  }, /*#__PURE__*/_react.default.createElement(_containers.Box, {
    size: "content",
    className: "items-center flex flex-col md:flex-row"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex-none"
  }, children && children.logo), /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-col md:flex-row flex-1 w-full justify-end"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "flex flex-grow-0 text-center my-4 md:my-0"
  }, children && children.content && /*#__PURE__*/_react.default.createElement(_containers.Grid, {
    className: (0, _classnames.default)("grid-cols-2 w-full gap-8 md:mx-0", `md:grid-cols-${_react.default.Children.count(children.content)}`)
  }, children.content)), copy && /*#__PURE__*/_react.default.createElement("p", {
    className: "text-center my-8 md:my-0 md:ml-16"
  }, copy))));
};

exports.Footer1 = Footer1;