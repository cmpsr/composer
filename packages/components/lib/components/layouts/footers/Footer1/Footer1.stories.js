"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withCopyContent = exports.withContent = exports.withCopy = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

var _primitives = require("../../../primitives");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Composer/Layout/Footer/Footer1',
  component: _.Footer1
};
exports.default = _default;

const withCopy = () => /*#__PURE__*/_react.default.createElement(_.Footer1, {
  copy: "\xA9 2020 Composer",
  style: {}
}, {
  logo: /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "https://www.google.com"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Asset, {
    asset: {
      title: '',
      url: 'https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg',
      contentType: 'image'
    }
  }))
});

exports.withCopy = withCopy;
withCopy.story = {
  name: 'With copy'
};

const withContent = () => /*#__PURE__*/_react.default.createElement(_.Footer1, {
  style: {}
}, {
  logo: /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "https://www.google.com"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Asset, {
    asset: {
      title: '',
      url: 'https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg',
      contentType: 'image'
    }
  })),
  content: [/*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "0"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link")), /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "1"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link")), /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "2"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link")), /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "3"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link"))]
});

exports.withContent = withContent;
withContent.story = {
  name: 'With content'
};

const withCopyContent = () => /*#__PURE__*/_react.default.createElement(_.Footer1, {
  copy: "\xA9 2020 Composer",
  style: {}
}, {
  logo: /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "https://www.google.com"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Asset, {
    asset: {
      title: '',
      url: 'https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg',
      contentType: 'image'
    }
  })),
  content: [/*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "0"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link")), /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "1"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link")), /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "2"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link")), /*#__PURE__*/_react.default.createElement(_primitives.Link, {
    url: "#",
    key: "3"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Typography, {
    tag: "span",
    type: _primitives.TypographyTypes.Link
  }, "Link"))]
});

exports.withCopyContent = withCopyContent;
withCopyContent.story = {
  name: 'With copy and content'
};