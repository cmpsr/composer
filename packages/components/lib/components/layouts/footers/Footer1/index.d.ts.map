{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../src/components/layouts/footers/Footer1/index.ts"],"names":[],"mappings":"AAAA,cAAc,WAAW,CAAC"}