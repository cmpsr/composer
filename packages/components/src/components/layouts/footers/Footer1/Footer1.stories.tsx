import React from 'react';
import { Footer1 } from '.';
import {
  Link,
  Asset,
  Typography,
  TypographyTypes,
} from 'components/primitives';

export default {
  title: 'Composer/Layout/Footer/Footer1',
  component: Footer1,
};

export const withCopy = () => (
  <Footer1 copy="© 2020 Composer" style={{}}>
    {{
      logo: (
        <Link url="https://www.google.com">
          <Asset
            asset={{
              title: '',
              url:
                'https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg',
              contentType: 'image',
            }}
          />
        </Link>
      ),
    }}
  </Footer1>
);

withCopy.story = {
  name: 'With copy',
};

export const withContent = () => (
  <Footer1 style={{}}>
    {{
      logo: (
        <Link url="https://www.google.com">
          <Asset
            asset={{
              title: '',
              url:
                'https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg',
              contentType: 'image',
            }}
          />
        </Link>
      ),
      content: [
        <Link url="#" key="0">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
        <Link url="#" key="1">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
        <Link url="#" key="2">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
        <Link url="#" key="3">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
      ],
    }}
  </Footer1>
);

withContent.story = {
  name: 'With content',
};

export const withCopyContent = () => (
  <Footer1 copy="© 2020 Composer" style={{}}>
    {{
      logo: (
        <Link url="https://www.google.com">
          <Asset
            asset={{
              title: '',
              url:
                'https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg',
              contentType: 'image',
            }}
          />
        </Link>
      ),
      content: [
        <Link url="#" key="0">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
        <Link url="#" key="1">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
        <Link url="#" key="2">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
        <Link url="#" key="3">
          <Typography tag="span" type={TypographyTypes.Link}>
            Link
          </Typography>
        </Link>,
      ],
    }}
  </Footer1>
);

withCopyContent.story = {
  name: 'With copy and content',
};
