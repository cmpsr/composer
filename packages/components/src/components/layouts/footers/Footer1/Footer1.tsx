import React, { ReactElement } from "react";
import cn from "classnames";

import { Box, Grid } from "components/containers";
import { getClassesFromStyle } from "utils";

import { FooterStyle } from "types";

export const FOOTER1_DEFAULT_TEST_ID = "footer1";

type Props = {
  copy?: string;
  className?: string;
  children?: {
    logo?: ReactElement;
    content?: ReactElement | ReactElement[];
  };
  style: FooterStyle;
  testId?: string;
};

export const Footer1 = ({
  className,
  copy,
  children,
  style,
  testId = FOOTER1_DEFAULT_TEST_ID,
}: Props) => {
  const colorClasses = getClassesFromStyle(style.background);
  return (
    <Box
      size="full"
      className={cn("py-8", className, colorClasses)}
      testId={testId}
    >
      <Box size="content" className="items-center flex flex-col md:flex-row">
        <div className="flex-none">{children && children.logo}</div>
        <div className="flex flex-col md:flex-row flex-1 w-full justify-end">
          <div className="flex flex-grow-0 text-center my-4 md:my-0">
            {children && children.content && (
              <Grid
                className={cn(
                  "grid-cols-2 w-full gap-8 md:mx-0",
                  `md:grid-cols-${React.Children.count(children.content)}`
                )}
              >
                {children.content}
              </Grid>
            )}
          </div>
          {copy && <p className="text-center my-8 md:my-0 md:ml-16">{copy}</p>}
        </div>
      </Box>
    </Box>
  );
};
