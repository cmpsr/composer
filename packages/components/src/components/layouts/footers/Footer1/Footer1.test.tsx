import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Footer1, FOOTER1_DEFAULT_TEST_ID } from './Footer1';

describe('Footer1', () => {
  const testStyle = { background: { default: { bg: 'white' } } };
  it('should match snapshot', () => {
    const footer = renderer.create(<Footer1 style={testStyle} />);
    const tree = footer.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(<Footer1 style={testStyle} />);
    const footer = getByTestId(FOOTER1_DEFAULT_TEST_ID);
    expect(footer).toBeInTheDocument();
  });
  it('should render style', () => {
    const { getByTestId } = render(<Footer1 style={testStyle} />);
    const footer = getByTestId(FOOTER1_DEFAULT_TEST_ID);
    expect(footer.classList).toContain('bg-white');
  });
  it('should render copy', () => {
    const testCopy = 'Test copy';
    const { getByText } = render(<Footer1 style={testStyle} copy={testCopy} />);
    const copy = getByText(testCopy);
    expect(copy).toBeInTheDocument();
  });
  it('should render children logo', () => {
    const childrenLogoTestId = 'Children-Logo-TestId';
    const renderedChildren = {
      logo: <p data-testid={childrenLogoTestId}>Logo</p>,
      content: <div>Children</div>,
    };
    const { getByTestId } = render(
      <Footer1 style={testStyle}>{renderedChildren}</Footer1>,
    );
    const childrenLogo = getByTestId(childrenLogoTestId);
    expect(childrenLogo).toBeInTheDocument();
    expect(childrenLogo.parentElement.classList).toContain('flex-none');
  });
  it('should render children with appropriate style', () => {
    const childrenTestId = 'Children-TestId';
    const renderedChildren = {
      logo: <p>Logo</p>,
      content: <div data-testid={childrenTestId}>Children</div>,
    };
    const { baseElement, getByTestId } = render(
      <Footer1 style={testStyle}>{renderedChildren}</Footer1>,
    );
    const children = getByTestId(childrenTestId);
    expect(children).toBeInTheDocument();
    expect(baseElement.innerHTML).toContain('md:grid-cols-1');
  });
  it('should render array of children with appropriate style', () => {
    const childrenTestId = 'Children-TestId';
    const renderedChildren = {
      logo: <p>Logo</p>,
      content: [
        <div data-testid={childrenTestId + '1'} key="1">
          Children1
        </div>,
        <div data-testid={childrenTestId + '2'} key="2">
          Children2
        </div>,
      ],
    };
    const { baseElement, getAllByTestId } = render(
      <Footer1 style={testStyle}>{renderedChildren}</Footer1>,
    );
    const children = getAllByTestId(childrenTestId, { exact: false });
    expect(children.length).toBe(2);
    expect(baseElement.innerHTML).toContain('md:grid-cols-2');
  });
});
