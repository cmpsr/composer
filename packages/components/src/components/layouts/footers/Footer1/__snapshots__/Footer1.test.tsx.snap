// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer1 should match snapshot 1`] = `
<div
  className="w-full h-full py-8 bg-white"
  data-testid="footer1"
>
  <div
    className="w-full h-full max-w-content mx-auto items-center flex flex-col md:flex-row"
    data-testid="box"
  >
    <div
      className="flex-none"
    />
    <div
      className="flex flex-col md:flex-row flex-1 w-full justify-end"
    >
      <div
        className="flex flex-grow-0 text-center my-4 md:my-0"
      />
    </div>
  </div>
</div>
`;
